tests/linear_solve_test.py

```diff
@@ -104,8 +104,10 @@ def test_solve_dense_psd(self):
     x = linear_solve.solve_cholesky(matvec, b)
     x2 = jax.numpy.linalg.solve(A, b)
     x3 = linear_solve.solve_inv(matvec, b)
+    x4 = linear_solve.solve_qr(matvec, b)
```

Comment:
I forgot: could this be moved to solve_dense? 
QR does not require psd I think.

Reply:
:), I agree about not requiring QR, however, this is exactly what we are doing in LM. I add a test to solve_dense as well.